crates/red_knot_python_semantic/resources/mdtest/scopes/nonlocal.md

```diff
@@ -0,0 +1,45 @@
+# Nonlocal references
+
+## One level up
+
+```py
+def f():
+    x = 1
+    def g():
+        reveal_type(x)  # revealed: Literal[1]
+```
+
+## Two levels up
+
+```py
+def f():
+    x = 1
+    def g():
+        def h():
+            reveal_type(x)  # revealed: Literal[1]
+```
+
+## Skips class scope
+
+```py
+def f():
+    x = 1
+
+    class C:
+        x = 2
+        def g():
+            reveal_type(x)  # revealed: Literal[1]
+```
+
+## Skips annotation-only assignment
+
+```py
+def f():
+    x = 1
+    def g():
+        # it's pretty weird to have an annotated assignment in a function where the
```

Comment:
Should this be a TODO?

```suggestion
        # TODO: it's pretty weird to have an annotated assignment in a function where the
```

Reply:
FWIW, I intentionally didn't make it a TODO when I added it in the old test. It's in a gray area, but to me we should reserve TODO for things we definitely need to change, or at least revisit. I don't feel like it's clear that we need to change or revisit anything here. It's more like a musing that may be relevant to someone in the future considering this case.